### PR TITLE
Default Autotester EKAT BFB

### PR DESCRIPTION
**Pull request by tcclevenger**
Default `EKAT_DEFAULT_BFB=ON` for Autotester and simplify `ekat::view_reduction()`.

### CODE REVIEW

**Comment by bartgol** [comment]
I'm actually having second thoughts on whether this option should be in jenkins_common.sh. I think it's best to put it inside `test_all_scream.py`, as a common config option for all builds.

The reason is that, in the future, we _might_ add a Release build to our test-all-scream builds, with end-to-end runs against baselines, and either no unit tests or tolerance-based unit tests. In that case, we'd want to test _exactly_ what runs in production, so without the BFB behaviors. Adding the option to jenkins_common.sh would make _all_ the builds have such option, but adding it manually to the three builds, would allow a fourth build (the Release one) to be run without such option.

@jgfouca what do you think? Am I overthinking this?

**Comment by tcclevenger** [comment]
@bartgol @jgfouca I move the `EKAT_DEFAULT_BFB=ON` to `test_all_scream.py`. Let me know your thoughts. The other option is to put it in `jenkins_common.sh` (was 1st commit in the PR).

File: components/scream/scripts/test_all_scream.py
Comment (bartgol) [inline components/scream/scripts/test_all_scream.py:56]: I would not add it to the custom cmake opts, cause those are applied to all builds. I would instead add this to all dict entries below.

**Review by jgfouca — APPROVED** [review]
I approve pending @bartgol 's cmake_opts comment being addressed.

**Comment by tcclevenger** [comment]
@jgfouca @bartgol All tests pass, let me know if you have any other comments. If not, I'll need one more approval.

**Review by bartgol — APPROVED** [review]

**Pull request merged**
